### ISSUE TITLE
[Feature Request] Add a search option in select component. 

### ISSUE THREAD

**Issue by nayan1834** [nextui-org/nextui#3367]
### Is your feature request related to a problem? Please describe.

When dealing with a Select component that contains a large number of options (e.g., hundreds or thousands of items), it becomes challenging for users to find and select the desired option efficiently. Scrolling through a lengthy list is time-consuming and can lead to user frustration. 
Currently, the Select component lacks a built-in search or filter mechanism, which forces users to manually scroll through the entire list. This limitation significantly impacts user experience and efficiency, especially in applications where quick selection from large datasets is frequent.

### Describe the solution you'd like

I propose adding a search/filter functionality to the Select component. This feature would include:

- A search input field at the top of the dropdown list.
- Real-time filtering of the options as the user types.
- Highlighting of matched text in the filtered results.
- Clear indication when no results match the search criteria.
- The ability to disable/hide this feature for smaller lists where it's not needed.

### Describe alternatives you've considered

i am using some packages from npm
also reactprime has a component named multiselect. Which is great.

**Comment by SySagar** [comment]
If possible can I take up thi issues?

**Comment by nayan1834** [comment]
> If possible can I take up thi issues?

Sure.

**Comment by abhinandan-verma** [comment]
@nayan1834  I think the `autocomplete` component can fulfil your need up to some extent. For the searching part, I need confirmation from the team. Then I can confirm you.

**Comment by Shivansh-Verma7719** [comment]
@ysnozcn I agree please look into this, I also need to use multi select with a dropdown and I can't right now.

**Comment by GVALFER** [comment]
yep, I also need this component. 
Autocomplete do not fit for me... Also, works a little strange for a "select"
